xbmc/windowing/android/WinSystemAndroid.cpp

```diff
@@ -219,66 +217,48 @@ void CWinSystemAndroid::UpdateResolutions(bool bUpdateDesktopRes)
 
 void CWinSystemAndroid::OnTimeout()
 {
-  // We don't trigger OnResetDisplay if we wait for HDMI connect
-  {
-    CSingleLock lock(m_resourceSection);
-    if (m_dispResetState & RESET_WAIT_HDMIPLUG)
-    {
-      // Let HDMI_PLUG trigger the reset
-      m_dispResetState &= ~RESET_WAIT_HDMIPLUG;
-      return;
-    }
-  }
-  SetHDMIState(HDMI_STATE_CONNECTED);
+  SetHDMIState(true);
```

Comment:
I am not sure until today, if this thread, which executes the OnTimeout has additional side-effects. Other platforms reset that stuff on their mainloop. On the other hand you properly take the resource lock below and it should work.